src/NHibernate/Linq/NhLinqExpression.cs

```diff
@@ -34,7 +36,8 @@ public class NhLinqExpression : IQueryExpression
 
 		private readonly Expression _expression;
 		private readonly IDictionary<ConstantExpression, NamedParameter> _constantToParameterMap;
-
+		private Lazy<QueryModelWithExtractedOptions> _queryModelAndOptions;
```

Comment:
I don't understand why do you need this lazy.

Reply:
The idea was to make sure that the extraction process was run as few times as possible. When the query plan is cached the first Lazy instance will be invoked, since Translate will not be called, but as soon as it is called, the Lazy is replaced with a direct reference. Perhaps it wasn't necessary. It would also be good to avoid the double Translate.